arm-customer-insights/2017-04-26/swagger/customer-insights.json

```diff
@@ -0,0 +1,5664 @@
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CustomerInsights/hubs/{hubName}/kpi/{kpiName}": {
+      "put": {
+        "tags": [
+          "Kpi"
+        ],
+        "operationId": "Kpi_CreateOrUpdate",
+        "x-ms-examples": {
+          "Kpi_CreateOrUpdate": {
+            "$ref": "../examples/KpiCreateOrUpdate.json"
+          }
+        },
+        "description": "Creates a KPI or updates an existing KPI in the hub.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "name": "kpiName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the KPI.",
+            "pattern": "^[a-zA-Z][a-zA-Z0-9_]+$",
+            "minLength": 1,
+            "maxLength": 512
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/KpiResourceFormat"
+            },
+            "description": "Parameters supplied to the create/update KPI operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Successfully created the KPI.",
+            "schema": {
+              "$ref": "#/definitions/KpiResourceFormat"
+            }
+          },
+          "202": {
+            "description": "Accepted. The create KPI operation is accepted."
+          }
+        },
+        "x-ms-long-running-operation": true
+      },
+      "get": {
+        "tags": [
+          "Kpi"
+        ],
+        "operationId": "Kpi_Get",
+        "x-ms-examples": {
+          "Kpi_Get": {
+            "$ref": "../examples/KpiGet.json"
+          }
+        },
+        "description": "Gets a KPI in the hub.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "hubName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the hub."
+          },
+          {
+            "name": "kpiName",
+            "in": "path",
+            "requir
```

Comment:
Hi, this api is to get the images' upload URL. As you can see in the swagger it depends on the entityType, entityTypeName, relativePath, so the URL is not the same for all images in a hub, but it has nothing to do with the imageName. This API is in the ImagesController, I think that's why the person who developed this add a /images segment